core/app/models/spree/variant.rb

```diff
@@ -39,6 +39,7 @@ class Variant < Spree::Base
     has_many :images, -> { order(:position) }, as: :viewable, dependent: :destroy, class_name: "Spree::Image"
 
     has_many :prices,
+      -> { with_deleted },
```

Comment:
we really need to explain, why we use `with_deleted` all the time on the prices.